PRC History Review.js

```diff
@@ -0,0 +1,297 @@
+					if (issueInfoEl) {
+						const issueInfoArray = issueInfoEl.textContent.split('★');
+						if (issueInfoArray.length === 3) {
+							const volume = issueInfoArray[0];
+							newItem.volume = volume.toLowerCase().replace('volume', '').trim();
+							const issue = issueInfoArray[1];
+							newItem.issue = issue.toLowerCase().replace('number', '').trim();
+							const date = issueInfoArray[2];
+							newItem.date = ZU.strToISO(date);
+						}
+					}
+				}
+				else if (isSingleArticle) {
+					// if it's single article page, the issue info is two divs above the link div.
+					const issueInfoElIndex = index - 2;
+					if (issueInfoElIndex >= 0) {
+						const issueInfoEl = doc.querySelectorAll('h5')[issueInfoElIndex];
+						if (issueInfoEl) {
+							const issueInfoArray = issueInfoEl.textContent.split('★');
+							if (issueInfoArray.length === 3) {
+								const volume = issueInfoArray[0];
+								newItem.volume = volume.toLowerCase().replace('volume', '').trim();
+								const issue = issueInfoArray[1];
+								newItem.issue = issue.toLowerCase().replace('number', '').trim();
+								const date = issueInfoArray[2];
+								newItem.date = ZU.strToISO(date);
+							}
+						}
+					}
+				}
+				else {
+					// if it's homepage article, the issue info is in the previous div.
+					const lastIndex = index - 1;
+					if (lastIndex >= 0) {
+						const issueInfoEl = articleLinkEls[lastIndex];
+						const issueInfoArray = issueInfoEl.textContent.split('★');
+						if (issueInfoArray.length === 2) {
+							const issue = issueInfoArray[0];
+							newItem.issue = issue.toLowerCase().replace('number', '').trim();
+							const date = issueInfoArray[1];
+							newItem.date = ZU.strToISO(date);
+						}
+					}
+				}
+
+				// add author
+				if (isIssue) {
+					const authorEl = articleLinkEls[index + 1];
+					if (authorEl) {
+						// in case there are multiple authors
+						const authorTexts = authorEl.textContent.trim().split(' and ');
+						authorTexts.forEach((authorText) => {
+							const authorName = authorText.split(',')[0];
+							if (authorName) {
+								newItem.creators.push(ZU.cleanAuthor(authorName, 'author', false));
+							}
+						});
+					}
+				}
+
+				if (isSingleArticle) {
+					const rawAuthorText = linkEl.textContent.split(',')[0];
+					if (rawAuthorText) {
+						const authorTexts = rawAuthorText.trim().split(' and ');
+						authorTexts.forEach((authorText) => {
+							const authorName = authorText.split(',')[0];
+							if (authorName) {
+								newItem.creators.push(ZU.cleanAuthor(authorName, 'author', false));
+							}
+						});
+					}
+				}
+
+				// Download pdf
+				const pdfUrl = url;
+				if (pdfUrl && pdfUrl.toLowerCase().endsWith('.pdf')) {
+					newItem.attachments.push({
+						url: pdfUrl,
+						mimeType: "application/pdf",
+					});
+				}
+				newItem.complete();
+			}
+		}
+	});
+}
+
+// helper functions.
+function checkIfLinkIsPdf(url) {
```

Comment:
Maybe better to rename this and the following method `isPdfUrl` and `isHomePageUrl` to follow convention.

Reply:
Done